common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -93,7 +93,10 @@ protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBO
     override val sentinelledLogs = false
 }
 
-protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
+/**
+ * add temporary field that holds the "fixed" names for components where the '_' is replaced by the user's namespace
+ */
+protected[core] case class SequenceExec(code: String, components: Vector[String], fixedComponents: Option[Vector[String]] = None) extends Exec(Exec.SEQUENCE) {
```

Comment:
I just quickly check by storing fixedComponents in components that Perry's code fails. Since at some point I'll remove Perry's code, I'll leave this as is and open an issue to keep track of cleanup stuff.